internal/handlers/pg/pgdb/query.go

```diff
@@ -271,12 +271,18 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 					case *types.Document, *types.Array, types.Binary,
 						types.NullType, types.Regex, types.Timestamp:
 						// type not supported for pushdown
-					case float64, string, types.ObjectID, bool, time.Time, int32, int64:
+					case float64, types.ObjectID, bool, time.Time, int32, int64:
 						// Select if value under the key is equal to provided value.
 						sql := `_jsonb->%[1]s @> %[2]s`
 
 						filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next()))
 						args = append(args, rootKey, string(must.NotFail(pjson.MarshalSingleValue(v))))
+					case string:
+						// Select if value under the key is equal to provided value.
+						sql := `_jsonb->%[1]s @> %[2]s`
```

Comment:
Please try `%[2]s::jsonb` or something similar

Reply:
I've tried that and a couple of things like ::text etc. PostgreSQL does casting to jsonb automatically, so the error is the same with `::jsonb` and without it.

I've tried to debug how pgx behave with different scenarios, and check generated queries.

From my understanding: you can pass numbers either as strings or as numbers. If you want to specify that you pass text, you need to add another quotes inside string.

So, from Go perspective we can pass to pgx both `"42"` and `42` to represent a number.
But for strings we need to do `"\"foo\""`.